nemo/collections/nlp/parts/nlp_overrides.py

```diff
@@ -556,8 +580,19 @@ def load_checkpoint(self, checkpoint_path: Union[str, Path], load_optimizer_stat
             # Check whether to load optim states
             if load_optimizer_states:
                 checkpoint['optimizer_states'] = [self.optimizer_sharded_state_dict(is_loading=True)]
+
+            # Check if the model has a separate LM head stage
+            def is_lm_head_separate_stage(module=None, state_dict=None):
+                if module is None or state_dict is None:
+                    return False
+                if len(state_dict) > 1:
```

Comment:
Another question - Is it possible to check [self.post_process and self.post_layer_norm](https://github.com/NVIDIA/Megatron-LM/blob/main/megatron/core/transformer/transformer_block.py#L248) for the case where there's a final_layernorm?